### PR TITLE
feat: add basic re-routeable logger

### PR DESCRIPTION
**Pull request by FrogTheFrog**
## Description
Add a basic re-routable logger (with the intention of connecting to Sunshine later on) that follows
the `BOOST_LOG(info) << ...` pattern for a drop-in replacement when porting the code.

Additionally C++ is bumped to 20 and some other minor cleanup was performed.

## Type of Change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Dependency update (updates to dependencies)
- [ ] Documentation update (changes to documentation)
- [ ] Repository update (changes to repository files, e.g. `.github/...`)

## Checklist
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated the in code docstring/documentation-blocks for new or existing methods/components

## Branch Updates
LizardByte requires that branches be up-to-date before merging. This means that after any PR is merged, this branch
must be updated before it can be merged. You must also
[Allow edits from maintainers](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [x] I want maintainers to keep my branch updated

### CODE REVIEW

**Review by ReenigneArcher — COMMENTED** [review]
My only question would be, should there be a fatal/critical level?

Otherwise, everything else looks fine (once the one test is fixed)

**Comment by FrogTheFrog** [comment]
> My only question would be, should there be a fatal/critical level?

I was actually on a fence about it, fatal is probably not going to be used as for `Critical`... I'm using the Sunshine log pattern here, so `Critical == Error` in my case. What's missing is a `Warning`, but that also is not used.

Should we add them already or only when we need it?

**Comment by ReenigneArcher** [comment]
> Should we add them already or only when we need it?

I think it's probably likely that there will be warning level messages. Probably easier to add it in now?

**Comment by FrogTheFrog** [comment]
> > Should we add them already or only when we need it?
> 
> I think it's probably likely that there will be warning level messages. Probably easier to add it in now?

Just warning, no Fatal?

**Comment by ReenigneArcher** [comment]
> Just warning, no Fatal?

Yea, that's fine.

**Review by ReenigneArcher — APPROVED** [review]

File: CMakeLists.txt [hunk 1]
```diff
@@ -31,7 +31,7 @@ else()
 endif()
 
 # glob src
-file(GLOB_RECURSE DD_TARGET_FILES src/*.cpp)
+file(GLOB_RECURSE DD_TARGET_FILES src/*.h src/*.cpp)
```
Comment (ReenigneArcher) [inline CMakeLists.txt:34]: Per https://github.com/LizardByte/libdisplaydevice/pull/10#discussion_r1567371561

We should probably not add `*.h` here.

```suggestion
file(GLOB_RECURSE DD_TARGET_FILES src/*.cpp)
include_directories("${CMAKE_SOURCE_DIR}")
```

**Pull request merged**